placement/types.go

```diff
@@ -85,6 +85,12 @@ type Host interface {
 	fmt.Stringer
 	ID() string
 	Rack() string
+	Zone() string
+}
+
+// NewHost returns a Host
+func NewHost(id, rack, zone string) Host {
```

Comment:
Perhaps move the implementation to separate files and only keep type definitinos in `types.go` ?